### PR TITLE
Add runtime sanitizer configuration file

### PR DESCRIPTION
**Pull request by peddie**
This was available in starling long ago but hasn't made it to other
repos.

### CODE REVIEW

File: SanitizeTargets.cmake [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Runtime analysis using Clang sanitization flags.
+
+option(ENABLE_SANITIZERS "Enable sanitizers." OFF)
```
Comment (isaactorz) [inline SanitizeTargets.cmake:3]: Small nit but do we need this option? I think the way it's set up currently if I wanted to turn on UB sanitizing for instance i'd always have to do:
```sh
cmake -DENABLE_SANITIZERS=ON -DSANITIZE_UNDEFINED=ON ..
```

Seems a little redundant.

**Review by isaactorz — COMMENTED** [review]
Another nit

Maybe we should prefix these options with `SWIFT` or something?

Would reduce the risk of colliding with similarly named options in other codebases.

File: SanitizeTargets.cmake (hunk 1 above)
Comment (peddie) [reply SanitizeTargets.cmake:3]: Sure, I'll remove the top-level flag.  I just took the file from starling.

**Comment by peddie** [comment]
Seems a bit of a shame to stick `SWIFT` in front of everything, but on the other hand, I could see myself wasting a lot of time on subtle conflicts with a submodule somewhere without it.  Done

**Review by isaactorz — APPROVED** [review]

**Pull request merged**